runtime/js/fs_fake.js

```diff
@@ -81,6 +81,17 @@ class MlFakeDevice {
     }
   }
 
+  rename(oldname, newname) {
+    if (this.exists(newname))
+      caml_raise_sys_error(this.nm(newname) + " : file already exists");
+    if (!this.exists(oldname))
+      caml_raise_sys_error(this.nm(oldname) + " : no such file or directory");
+    if (this.is_dir(oldname))
+      caml_raise_sys_error(this.nm(oldname) + " : not a file");
```

Comment:
The doc says:
> Rename a file or directory.  [rename oldpath newpath] renames the
    file or directory called [oldpath], giving it [newpath] as its new name,
    moving it between (parent) directories if needed. 

We shouldn't fail on directory.

You should probably add tests in `compiler/tests-compiler/sys_fs.ml`